lab3/client/src/reducer.js

```diff
@@ -0,0 +1,42 @@
+import constants from './constants';
+import Generator from './models/generator';
+
+export default function reducer (state, action) {
+	switch (action.type) {
+		case constants.actions.EXAMPLE:
+			state.example = action.payload;
+			return state;
+		case constants.actions.INCREMENT:
+			state.counter += action.payload;
+			return state;
+		break;
+		case constants.actions.BUY_GENERATOR:
+			const currentGenerator = action.generatorClicked || 0;
+
+			state.counter -= state.generators[currentGenerator].baseCost;
+			state.generators[currentGenerator].quantity += 1;
+
+
+			const gModel = new Generator(state.generators[currentGenerator]);
+			state.generators[currentGenerator].baseCost = Math.ceil(gModel.getCost());
```

Comment:
don't change the baseCost, the baseCost should always remain the same and the `getCost` function will give the computed value based on its quantity and base cost.